src/Web/Telegram/API/Bot/API.hs

```diff
@@ -166,8 +172,11 @@ forwardMessage :: Token -> ForwardMessageRequest -> IO (Either ServantError Mess
 forwardMessage = run forwardMessage_
 
 -- | Use this method to send photos. On success, the sent 'Message' is returned.
-sendPhoto :: Token -> SendPhotoRequest -> IO (Either ServantError MessageResponse)
-sendPhoto = run sendPhoto_
+sendPhoto :: Token -> SendPhotoRequest (Either Text FileUpload) -> IO (Either ServantError MessageResponse)
+sendPhoto token sendPhotoReq =
+  case photo_photo sendPhotoReq of
+    Left  fileId     -> run sendPhotoFileId_     token (sendPhotoReq { photo_photo = fileId })
+    Right fileUpload -> run sendPhotoFileUpload_ token (sendPhotoReq { photo_photo = fileUpload })

```

Comment:
If possible I would split `sendPhoto` method in two methods instead of using `Either`

```
sendPhoto :: Token -> SendPhotoRequest FileUpload -> IO (Either ServantError MessageResponse)
sendPhotoById :: Token -> SendPhotoRequest Text -> IO (Either ServantError MessageResponse)
```